android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/WebIntentFragment.java

```diff
@@ -0,0 +1,41 @@
+package com.shopify.unity.buy;
+
+import android.app.Fragment;
+import android.net.Uri;
+import android.support.customtabs.CustomTabsIntent;
+
+public class WebIntentFragment extends Fragment {
+
+    private boolean didShowIntent = false;
+    private WebIntentListener listener;
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (didShowIntent) {
+            notifyListenerOnClose();
+        }
+    }
+
+    public void setListener(WebIntentListener listener) {
+        this.listener = listener;
+    }
+
+    public void launchUrl(String url) {
+        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
+        CustomTabsIntent customTabsIntent = builder.build();
+
+        didShowIntent = true;
+        customTabsIntent.launchUrl(getActivity().getApplicationContext(), Uri.parse(url));
```

Comment:
Depends, but if it's related to UI stuff (like Android Pay requires Activity context is it shows UI). So I guess here it's fine to get activity as context.

Agh one note here, as this is a public method make sure about lifecycle, add verification that fragment is not detached from the activity